src/Microsoft.Health.Fhir.SqlServer/Features/Schema/ResourceVersion.cs

```diff
@@ -0,0 +1,36 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Schema
+{
+    public static class ResourceVersion
+    {
+        private static bool versionIsLong = true;
+
+        public static int GetVersion(SqlDataReader reader, int index)
```

Comment:
Curious also why we can't leverage the SchemaInformation for this? e.g.

```csharp
public static long GetVersion(SqlDataReader reader, int index, SchemaInformation schemaInformation)
{
    if (schemaInformation.Current >= SchemaVersionConstants.VersionAsLong)
    {
        return reader.GetInt64(index);
    }
    else
    {
        return reader.GetInt32(index);
    }
}
```

Reply:
Discussed that option as well. With that approach, there might be a possibility of a gap between actual schema update and schemainformation object reflecting that.